### PR TITLE
Typed supervision did not handle stopped behaviors correctly 

### LINKED ISSUE #25156: restarting supervision strategy does not work with stopped initial behavior
```scala
spawn(
  Behaviors.supervise(Behaviors.stopped[Command])
    .onFailure(SupervisorStrategy.restart),
  "actor",
)
```
should produce actor which is immeditately stopped – it should behave as

```scala
spawn(Behaviors.stopped[Command], "actor")
```

and

```scala
spawn(Behaviors.setup[Command] { _ => Behaviors.stopped }, "actor")
```

This is a corner case, but supporting it may be useful when actor needs to process several commands when it starts / after restart.

ScalaTest:

```scala
class StopTest extends FunSuite with ActorTestKit {
  import StopTest._

  test("pass") {
    val actor = spawn(Behaviors.stopped[Command])
    TestProbe().expectTerminated(actor, 3.second)
  }

  test("fail") {
    val actor = spawn(
      Behaviors.supervise(Behaviors.stopped[Command])
        .onFailure(SupervisorStrategy.restart)
    )
    TestProbe().expectTerminated(actor, 3.second)
  }
}

object StopTest {
  sealed trait Command
}
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #25156

### CODE REVIEW

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala [hunk 1]
```diff
@@ -690,4 +690,58 @@ class SupervisionSpec extends ActorTestKit with TypedAkkaSpecWithShutdown {
...
+  )
+
+  allStrategies.foreach { strategy ⇒
+
+    s"Supervision with the strategy $strategy" should {
+
+      "that is initially stopped should be stopped" in {
+        val actor = spawn(
+          Behaviors.supervise(Behaviors.stopped[Command])
+            .onFailure(SupervisorStrategy.restart)
```
Comment (chbatey) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala]: do you not want to use `strategy` here?

**Review by chbatey — APPROVED** [review]
One comment about test then LGTM

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala (hunk 1 above)
Comment (johanandren) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala]: ups

Comment (patriknw) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala]: I have fixed this. No permission to push to your branch, so continued in https://github.com/akka/akka/pull/25334

Resume was missing.

**Comment by patriknw** [comment]
Continued in https://github.com/akka/akka/pull/25334

**Pull request closed without merging**